src/Peachpie.CodeAnalysis/Utilities/BitMask.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Peachpie.CodeAnalysis.Utilities
+{
+    /// <summary>
+    /// A general structure to record bit information about indexed entities in flow analysis.
+    /// </summary>
+    /// <remarks>
+    /// The minimum and initial lattice element is 0, the maximum is ~0. Indices outside the scope of ulong (64)
+    /// are regarded as present in the set.
+    /// </remarks>
+    internal struct BitMask
```

Comment:
isn't there enough BitMasks already?

Reply:
I knew only about `TypeRefMask`, which is specialized for type references. But as you mentioned it, I found out that there is also `PhpSignatureMask`. Should I use that? From its name and location it seems to be meant for usage in PHP symbols. I haven't found any other type (apart from directly using `ulong`).

Reply:
can be unified ... for now, maybe just use `ulong` in the analysis ? is 64 bits enough for this purpose?

Reply:
Yes, 64 bits are enough for both `ParameterAnalysis` and `CopyAnalysis` (according to running them in WordPress). The purpose of `BitMask` is to encapsulate the manipulation with particular bits and handling overflowing indices (any item with index >= 64 is understood as being in the mask - the same behaviour as in `TypeRefMask`). The code which uses `BitMask` seems clearer to me than if we used `ulong` instead. Do we really want to get rid of it?